### ISSUE TITLE
Socket.IO adapter doesn't clean up on disconnect.

### ISSUE THREAD

**Issue by Kledal** [thoov/ember-websockets#51]
The Socket.IO adapter doesn't clean up on the disconnect event. This means that all the events that are created inside the ```on('connect')``` is going to be fired multiple times, because they are going to be created again each time the app reconnects.

A simple example:
1. Connect ember application to a server(node etc.) with the socket.io adapter.
2. put a console.log inside an event etc. ```on('welcome')```
3. Fire the welcome event each time the clients connect.
4. Restart node server
5. Watch N+1 console.log each time you reconnect

TLDR; This problem isn't that ```on('connect')``` is fired multiple times. But that the previous created listeneres inside the connect event isn't destroyed on disconnect.

**Comment by thoov** [comment]
@Kledal thanks for finding this. I am looking into this

**Comment by thoov** [comment]
@Kledal So I tried to follow your instructions and this is what I have:

```js
import Ember from 'ember';

export default Ember.Controller.extend({
  socketService: Ember.inject.service('socket-io'),

  init: function() {
    this._super.apply(this, arguments);

    var socketIO = this.get('socketService').socketFor('http://localhost:9001/');

    socketIO.on('welcome', function() {
      console.log('Welcome event fired');
    }, this);
  }
});
```

And then I have this node file:
```js
var app = require('http').createServer(function() {});
var io  = require('socket.io')(app);

app.listen(9001);

io.on('connection', function (socket) {
  socket.emit('welcome');
});
```

However when I restart the node server I am not seeing the duplicated messages issue. Can let me know what i am doing differently then what you are doing.

**Comment by Kledal** [comment]
Thanks for your time @thoov. Your small example made me realise what the problem maybe is.

I followed the documentation that gives the following example(look a code snippet below) on how to connect with Socket.IO. But your code example doesn't have the ```on('welcome')``` inside a ```on('connect')``` like mine do. 

So I think that this is all about that the README is wrong.
```js
/*
    * 3) Define any event handlers
    */
    socket.on('connect', function() {

      /*
      * There are 2 ways to send messages to the server: send and emit
      */
      socket.send('Hello World');

      socket.on('message', this.onMessage, this);

      socket.on('myCustomNamespace', function() {
        socket.emit('anotherNamespace', 'some data');
      }, this);
    }, this);
```

**Comment by Kledal** [comment]
I have just made a pull request which fixes the documentation.
